12_findTheOldest/solution/findTheOldest-solution.js

```diff
@@ -1,19 +1,17 @@
-const findTheOldest = function (array) {
-  return array.reduce((oldest, currentPerson) => {
-    const oldestAge = getAge(oldest.yearOfBirth, oldest.yearOfDeath);
-    const currentAge = getAge(
-      currentPerson.yearOfBirth,
-      currentPerson.yearOfDeath
-    );
-    return oldestAge < currentAge ? currentPerson : oldest;
-  });
+const getAge = function (person) {
+  // The nullish coalescing assignment operator
+  // evaluates to the right hand side if the left hand side is undefined
+  // Else it evaluates to the left hand side
+  person.yearOfDeath ??= new Date().getFullYear();
+
+  return person.yearOfDeath - person.yearOfBirth;
 };
 
-const getAge = function (birth, death) {
-  if (!death) {
-    death = new Date().getFullYear();
-  }
-  return death - birth;
+const findTheOldest = function (people) {
+  // Sort the people from oldest to youngest
+  return people.sort(
+    (person, nextPerson) => getAge(nextPerson) - getAge(person),
+  )[0];
```

Comment:
2 things:
1. Given this point in the curriculum, I'm not a particular fan of the `fn()[i]` pattern, and feel it would be more appropriate for learners to see things written more explicitly.
2. `people` will contain a reference to the array object passed in, and `sort` is a mutating method so while it happens to pass these tests, is not actually an appropriate solution. `toSorted` should be used here.

e.g.
```js
const findTheOldest = function (people) {
  const sortedPeopleOldestFirst = people.toSorted(
    (person, nextPerson) => getAge(nextPerson) - getAge(person)
  );
  const oldestPerson = sortedPeopleOldestFirst[0];
  return oldestPerson;
};
```
The `const oldestPerson =` line is definitely optional and you can just return `sortedPeopleOldestFirst[0]` but I like the further explicitness for learning sake.

Reply:
Definitely agree on the explicitness, it can be harder to digest code that uses less variables!